azext_iot/product/providers/auth.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrest.authentication import Authentication
+
+
+class AICSAuthentication(Authentication):
+    def __init__(self, cmd, base_url):
+        self.cmd = cmd
+        self.base_url = base_url
+
+    def signed_session(self, session=None):
+        """
+        Create requests session with SAS auth headers.
+
+        If a session object is provided, configure it directly. Otherwise,
+        create a new session and return it.
+
+        Returns:
+            session (): requests.Session.
+        """
+
+        return self.refresh_session(session)
+
+    def refresh_session(
+        self, session=None,
+    ):
+        """
+        Refresh requests session with SAS auth headers.
+
+        If a session object is provided, configure it directly. Otherwise,
+        create a new session and return it.
+
+        Returns:
+            session (): requests.Session.
+        """
+
+        session = session or super(AICSAuthentication, self).signed_session()
+        session.headers["Authorization"] = self.generate_token()
+        return session
+
+    def generate_token(self):
+        from azure.cli.core._profile import Profile
+
+        profile = Profile(cli_ctx=self.cmd.cli_ctx)
+        creds, subscription, tenant = profile.get_raw_token()
```

Comment:
No resource Id (audience claim on the receiving end) is needed?

Reply:
We haven't needed it on any of the AICS environments.

Reply:
This is weird because Matt was saying they accept certain audiences in the tokens.

Reply:
Resolved - if no resource Id is explicitly added, then https://management.core.windows.net/ is used which Matt said is an allowed audience.